CRM/Contribute/BAO/Contribution.php

```diff
@@ -179,6 +179,10 @@ public static function add(&$params) {
      isset($params['financial_type_id']))) {
       $params['tax_amount'] = $taxAmount;
     }
+    // there are cases when tax_amount could be NULL (especially when creating a template for recurring contribution)
+    if (isset($params['tax_amount']) && empty($params['tax_amount'])) {
+      $params['tax_amount'] = $taxAmount;
+    }  
```

Comment:
I'm not sure I understand this code. If you're checking for null, isset will return false and empty will return true and the condition will fail.

Reply:
we should cast null to 0 I think

Reply:
Here we are fetching previous contribution record to create template. So value is NULL (from DB). Hence. isset passes ad empty is true